elasticsearch-transport/lib/elasticsearch/transport/transport/base.rb

```diff
@@ -42,7 +42,8 @@ def initialize(arguments={}, &block)
           @logger      = options[:logger]
           @tracer      = options[:tracer]
 
-          @sniffer     = options[:sniffer_class] ? options[:sniffer_class].new(self) : Sniffer.new(self)
+          sniffer_class = options[:sniffer_class] ? options[:sniffer_class] : Sniffer
+          @sniffer     = sniffer_class.new(self, logger)
```

Comment:
With the accessor it's the same thing though. I don't see the difference, but I can make the change if you feel strongly here.